### PR TITLE
Improved Sealing Key Handlers

### PR DESCRIPTION
**Pull request by bbengfort**
This PR provides interfaces for managing sealing keys and sealing key exchanges and implements the creation and selection of sealing keys from x509 certificates and RSA keys. 

Still TODO:

- [x] make story for documentation
- [x] address review comments
- [x] tests for exchange keys
- [x] tests for parsing key exchange data

### CODE REVIEW

File: pkg/trisa/keys/certs.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+// PublicKeySignature returns a unique identifier that can be used to manage public keys
+// and associate them with their counterpart private keys for unsealing.
+func (c *Certificate) PublicKeySignature() (string, error) {
+	return signature.New(c.certs.PublicKey)
+}
+
+// Marshal returns a PEM encoded Certificate Block along with a PEM encoded private key
+// block if one is available. This data is useful for storing Keys on disk or in a
```
Comment (rebeccabilbro) [inline pkg/trisa/keys/certs.go:103]: maybe "for storing *one's own* Keys on disk" ?

File: pkg/trisa/keys/certs.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+// PublicKeyAlgorithm refers to the public key algorithm of the x509.Certificate and
+// may be used to determine which cipher to use. Typically is "RSA".
+func (c *Certificate) PublicKeyAlgorithm() string {
+	return c.certs.PublicKeyAlgorithm.String()
+}
+
+// PublicKeySignature returns a unique identifier that can be used to manage public keys
+// and associate them with their counterpart private keys for unsealing.
```
Comment (rebeccabilbro) [inline pkg/trisa/keys/certs.go:97]: Suggest:

// PublicKeySignature returns a unique identifier that can be used to manage the public keys
// of one's counterparties; 
// or to manage one's own keys locally and associate them with the matching private keys for unsealing

File: pkg/trisa/keys/certs.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+			return nil, err
+		}
+	}
+
+	return b.Bytes(), nil
+}
+
+// Unmarshal a PEM encoded Certificate Block and optionally a PEM encoded private key
+// block if one is available. Unmarshal is designed to perform the inverse functionality
+// of Marhshal - e.g. Unmarshal will load Marshaled data. However, Unmarshal may work on
```
Comment (rebeccabilbro) [inline pkg/trisa/keys/certs.go:131]: Marhshal => Marshal

File: pkg/trisa/keys/signature/signature.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+// New creates a public key signature for an *rsa.PublicKey, *ecdsa.PublicKey or
+// ed25519.PublicKey (or any key that can be marshalled by x509.MarshalPKIXPublicKey).
+// It returns the default SHA256 signature that the package recommends for public key
+// identification and matching.
+func New(pub interface{}) (_ string, err error) {
+	return Sign(pub, SHA256)
+}
+
+// Match determes if the public key signature matches the specified public key.
```
Comment (rebeccabilbro) [inline pkg/trisa/keys/signature/signature.go:38]: determes => determines

File: pkg/trisa/keys/signature/signature.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+	var sum []byte
+	if sum, err = Hash(pub, algorithm); err != nil {
+		return false
+	}
+
+	return bytes.Equal(sum, checksum)
+}
+
+// Sign creates a public key signature from a public key that can bbe marshalled as an
```
Comment (rebeccabilbro) [inline pkg/trisa/keys/signature/signature.go:53]: bbe => be

File: pkg/trisa/keys/signature/signature.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
+/*
+Package signature provides a mechanism for computing public key signatures, which are
+used to help identify public keys used in sealing TRISA envelopes and select the
+matching private key pair when a secure envelope is received.
+
```
Comment (rebeccabilbro) [inline pkg/trisa/keys/signature/signature.go:5]: Worth it to add a recommendation in line with the white paper w/r/t sealing key usage? E.g.

TRISA strongly recommends that sealing/unsealing keys be distinct from identity certificates for improved security and to support differing retirement criteria. Some organizations may choose to use unique sealing/unsealing keys with each unique counterparty.

**Review by rebeccabilbro — APPROVED** [review]
What a nice addition to the TRISA codebase @bbengfort! I know that this will be a great resource to TRISA users, and we should definitely remember to add some documentation to trisa.dev about key mgmt best practices! 

I've noted just a small handful of things I found on my review; I hope you find the feedback useful!

File: pkg/trisa/keys/certs.go (hunk 1 above)
Comment (bbengfort) [reply pkg/trisa/keys/certs.go:103]: Technically could also be used for storing counterparty public keys - but that's an edge use case. Updated to:

> This data is useful for storing Keys (particularly one's own private key pairs and sealing certificates) on disk or in a database, but should not be used to transfer or send keys.

Hopefully that helps clarify things a bit.

File: pkg/trisa/keys/certs.go (hunk 2 above)
Comment (bbengfort) [reply pkg/trisa/keys/certs.go:97]: Does this help clarify?

```
// PublicKeySignature returns a unique identifier that can be used to manage public keys
// and associate them with their counterpart private keys for unsealing or to identify
// the keys in use by the counterparty for creating and sealing secure envelopes.
```

**Pull request merged**